src/messaging/ExchangeContext.cpp

```diff
@@ -322,8 +323,8 @@ ExchangeContext::ExchangeContext(ExchangeManager * em, uint16_t ExchangeId, cons
 
     SetAckPending(false);
 
-    // Do not request Ack for multicast
-    SetAutoRequestAck(!session->IsGroupSession());
+    // Request Ack when MRP is allowed. For group and connection-oriented sessions, this should be not set.
```

Comment:
```suggestion
    // Try to use MRP by default, if it's possible over this session.
```

Reply:
Changing it to "Try to use MRP by default, if it is allowed over this session."